documentation/concepts/vaults.md

```diff
@@ -1,3 +1,30 @@
 ```yaml
 title: Vaults
 ```
+
+Various protocols, implemented in Ockam, depend on a variety of standard
+cryptographic primitives or building blocks. Depending on the environment,
+these building blocks may be provided by a software implementation or a
+cryptographically capable hardware component.
+
+In order to support a variety of cryptographically capable hardware we
+wish to maintain loose coupling between a protocol and how a specific
+building block is invoked in a specific hardware. This is achieved using
+an abstract `Vault` interface.
+
+A concrete implementation of the `Vault` interface is called an Ockam
+Vault. Over time there will be many such implementations.
+
+![Vaults](./assets/vaults/vaults.svg)
+
+The `Vault` interface exposes functions for generating and using
+cryptographic keys, secure random numbers, key derivation, hashing,
+massage authentication, digital signatures, Diffie-Hellman key exchange,
+authenticated encryption etc.
+
+Our general design principle here is to minimize cryptographic agility
+and is aim for
+[well oiled joints](https://www.imperialviolet.org/2016/05/16/agility.html).
```

Comment:
My suggestion is to add a sentence that gives an overview of well oiled joints for now, and expand upon this concept later, in the context of our specific application of it.  In the mean time I feel that the link is a distraction.

Reply:
noted.